### PR TITLE
CLDC-3135 Infer beds for 24/25 bedsits

### PR DESCRIPTION
**Pull request by natdeanlewissoftwire**

### CODE REVIEW

File: app/models/derived_variables/lettings_log_variables.rb [hunk 1]
```diff
@@ -82,6 +82,9 @@ def set_derived_fields!
         self.prevten = 30 if owning_organisation&.provider_type == "LA"
       end
     end
+    if is_bedsit? && form.start_year_after_2024?
+      self.beds = 1
+    end
```
Comment (kosiakkatrina) [inline app/models/derived_variables/lettings_log_variables.rb]: Would it make sense to group all the 2024 things together under a single if statement?

File: app/models/derived_variables/lettings_log_variables.rb [hunk 2]
```diff
@@ -168,6 +171,9 @@ def clear_inapplicable_derived_values!
       self.wchair = nil
       self.location_id = nil
     end
+    if unittype_gn_changed? && unittype_gn_was == 2 && form.start_year_after_2024?
+      self.beds = nil
+    end
```
Comment (kosiakkatrina) [inline app/models/derived_variables/lettings_log_variables.rb]: Would this make more sense with the reset fields methods? Like `clear_inapplicable_derived_values!`

Comment (natdeanlewissoftwire) [reply app/models/derived_variables/lettings_log_variables.rb]: This _is_ in `clear_inapplicable_derived_values`... Am I missing something?

File: app/models/derived_variables/lettings_log_variables.rb (hunk 1 above)
Comment (natdeanlewissoftwire) [reply app/models/derived_variables/lettings_log_variables.rb]: Sure, I'll nest this for future use

File: app/models/derived_variables/lettings_log_variables.rb (hunk 2 above)
Comment (kosiakkatrina) [reply app/models/derived_variables/lettings_log_variables.rb]: Oh, haha, I thought it was in `set_derived_fields!` together with the change above 🤦 My other comment now doesn't make sense either, ignore me

**Review by kosiakkatrina — APPROVED** [review]

**Pull request merged**